doctoc.sh

```diff
@@ -0,0 +1,180 @@
+#!/bin/bash
+#
+# Takes a mardown file in argument, checks if a table of content is already
+# present, create a new one if there is none or update the current one.
+#
+# The script uses the $start_toc and $end_doc as delimiters for the table of
+# content. It works only if there is one pair of opening/closing delimiters. If
+# it finds a correct single pair, the content inside is updated.
+#
+# Attributions: Updated from https://gitlab.com/pedrolab/doctoc.sh/tree/master,
+# which comes from
+# https://gist.github.com/meleu/57867f4a01ede1bd730f14b2f018ae89.
+#
+# The list of invalid chars come from https://github.com/thlorenz/anchor-\
+# markdown-header/blob/56f77a232ab1915106ad1746b99333bf83ee32a2/anchor-\
+# markdown-header.js#L25
+#
+
```

Comment:
it's a good practice to start the shell script with
```
set -e
set -u
```
so that it exists early when an error occurs